avocado/utils/network/interfaces.py

```diff
@@ -128,6 +128,65 @@ def add_ipaddr(self, ipaddr, netmask):
         except Exception as ex:
             raise NWException("Failed to add address {}".format(ex))
 
+    @property
+    def vlans(self):
+        list_of_vlan_ids = {}
+        for i in self.host.interfaces:
+            details = i._get_interface_details()
+            if self.name == details.get("link"):
+                list_of_vlan_ids[details.get('linkinfo').get('info_data').get('id')] = details.get("ifname")
```

Comment:
There is too much gets here, and since get can return `None` it is a good idea to do a "direct get" inside a try..except KeyError:

```python
try:
   vland_id = details['linkinfo']['info_data']['id']
except KeyError as exc:
   ....
```

Reply:
Agreed. Do you want this function to return a dict or a list like your example?